messaging/controller/processor/ephemeral_keys.go

```diff
@@ -0,0 +1,59 @@
+package processor
+
+import (
+	"crypto/ecdsa"
+	"math/rand"
+	"sync"
+
+	"github.com/status-im/status-go/crypto"
+	cryptotypes "github.com/status-im/status-go/crypto/types"
+)
+
+type EphemeralKeysManager struct {
+	capacity int
+	keys     map[string]*ecdsa.PrivateKey
+	mutex    sync.RWMutex
+}
+
+func NewEphemeralKeysManager(capacity int) *EphemeralKeysManager {
+	return &EphemeralKeysManager{
+		capacity: capacity,
+		keys:     make(map[string]*ecdsa.PrivateKey),
+	}
+}
+
+func (e *EphemeralKeysManager) GetPrivateKeyFor(key *ecdsa.PublicKey) *ecdsa.PrivateKey {
+	e.mutex.RLock()
+	defer e.mutex.RUnlock()
+
+	return e.keys[cryptotypes.EncodeHex(crypto.FromECDSAPub(key))]
+}
+
+func (e *EphemeralKeysManager) GenerateOrGetRandom() (*ecdsa.PrivateKey, error) {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
+	if len(e.keys) >= e.capacity {
+		return e.getRandom(), nil
+	}
+
+	privateKey, err := crypto.GenerateKey()
+	if err != nil {
+		return nil, err
+	}
+
+	e.keys[cryptotypes.EncodeHex(crypto.FromECDSAPub(&privateKey.PublicKey))] = privateKey
```

Comment:
If we never cleanup this map, then maybe it's simpler to pre-generate the list of keys in `NewEphemeralKeysManager`? And then just always take a random key?

Reply:
> It would require returning an error from the constructor, and consequently from the constructors that rely on it

well.. it's better to get the error during initialization rather than during message processing.
Especially for generating key, which I assume shouldn't really fail

> . I’ll keep it lazy, but I renamed it to `GetRandom` because lazy key generation should be an implementation detail.

yeah, this is good for now 👍